### PR TITLE
35 add delete button

### PR DESCRIPTION
**Pull request by tom08**

### CODE REVIEW

**Comment by shaneallgeier** [comment]
Can you add a `confirm()` dialog when clicking the delete button?

**Comment by tom08** [comment]
I was working on this when I found the browser testing bug that Austin is working on.  So as soon as we figure out/decide what to do with that I'll add that code to this branch.

File: exdb/static/exdb/js/edit_experience.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"use strict";
```
Comment (shaneallgeier) [inline exdb/static/exdb/js/edit_experience.js]: This file should be merged with `experience_approval.js` since they are the same file.

**Comment by enki211** [comment]
Since we are changing the status to canceled, we should probably remove that from the home screen so they can not see the 'deleted' experiences

File: exdb/templates/exdb/experience_approval.html [hunk 2]
```diff
@@ -40,7 +40,8 @@
         <div>{% trans "Add Comment" %}:</div>
         <div class="form-errors">{{ comment_form.message.errors }}</div>
         <div>{{ comment_form.message.label }}{{ comment_form.message }}</div>
-        <input type="submit" value="{% trans "deny" %}" name="deny" />
-        <input type="submit" value="{% trans "approve" %}" name="approve" />
+        <input type="submit" value={% trans "deny" %} name="deny" />
+        <input type="submit" value={% trans "approve" %} name="approve" />
+        <input type="submit" id="delete" value={% trans "delete" %} name="delete" />
```
Comment (shaneallgeier) [inline exdb/templates/exdb/experience_approval.html]: These three lines are missing quotation marks around the `value` values.

File: exdb/tests/browser_tests.py [hunk 3]
```diff
@@ -308,6 +337,36 @@ def test_load(self):
...
+    def test_confirm_if_delete(self):
+        e = self.create_experience('pe',
+                                   start=make_aware(datetime(2020, 1, 1, 1, 30), timezone=utc),
+                                   end=make_aware(datetime(2021, 1, 1, 1, 30), timezone=utc))
+        self.client.get(reverse('approval', args=[e.pk]))
+        d = self.driver.find_element(By.CSS_SELECTOR, '#delete')
+        d.click()
+        alert = False
+        if self.driver.switch_to_alert():
+            alert = True
```
Comment (shaneallgeier) [inline exdb/tests/browser_tests.py]: Why did you do this like this?

File: exdb/tests/browser_tests.py [hunk 4]
```diff
@@ -308,6 +337,36 @@ def test_load(self):
...
 
+class EditExperienceBrowserTest(DefaultLiveServerTestCase):
+
+    def test_confirm_if_draft_delete(self):
+        e = self.create_experience('dr',
+                                   start=make_aware(datetime(2020, 1, 1, 1, 30), timezone=utc),
+                                   end=make_aware(datetime(2021, 1, 1, 1, 30), timezone=utc))
+        self.client.get(reverse('edit', args=[e.pk]))
+        d = self.driver.find_element(By.CSS_SELECTOR, '#delete')
+        d.click()
```
Comment (shaneallgeier) [inline exdb/tests/browser_tests.py]: The tests still pass even when I comment out this line. Something is amiss.

File: exdb/tests/browser_tests.py [hunk 5]
```diff
@@ -308,6 +337,51 @@ def test_load(self):
...
+
+        self.assertTrue(urls_equal, "The browser should stayed at the same url.")
+        self.assertFalse(exp_canceled, "The browser should have aborted the delete.")
+
+    def test_confirm_delete(self):
+        urls_equal, exp_canceled = self.delete_confirm(True)
+
+        self.assertFalse(urls_equal, "The browser should have went elsewhere.")
+        self.assertTrue(exp_canceled, "The browser should have continued with the delete.")
+
```
Comment (aaaustin10) [inline exdb/tests/browser_tests.py]: Please make sure to run verify at some point, my bad.

**Comment by tom08** [comment]
I approve of Austin's work.

File: exdb/tests/browser_tests.py [hunk 6]
```diff
@@ -10,6 +10,10 @@
 import selenium
 from selenium import webdriver
 from selenium.webdriver.common.by import By
+from selenium.webdriver.common.alert import Alert
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import NoAlertPresentException, NoSuchElementException
```
Comment (tom08) [inline exdb/tests/browser_tests.py]: I will take out these four unused imports in the next commit.

File: exdb/tests/browser_tests.py (hunk 4 above)
Comment (tom08) [inline exdb/tests/browser_tests.py]: I am copying the logic from ExperienceApprovalBrowserTest to this test as well in the next commit.

**Comment by shaneallgeier** [comment]
I approve.

File: exdb/tests/browser_tests.py [hunk 7]
```diff
@@ -343,14 +339,33 @@ def test_load(self):
...
+        ending_url = self.driver.current_url
+
+        urls_equal = starting_url == ending_url
+        exp_cancelled = Experience.objects.get(pk=e.pk).status == 'ca'
+        return urls_equal, exp_cancelled
+
+    def test_confirm_dont_delete(self):
+        urls_equal, exp_cancelled = self.delete_confirm(False)
+
+        self.assertTrue(urls_equal, "The browser should stayed at the same url.")
```
Comment (enki211) [inline exdb/tests/browser_tests.py]: should have?

**Comment by enki211** [comment]
Looks good

**Comment by shaneallgeier** [comment]
I approve again

**Comment by enki211** [comment]
Still looks good

**Pull request merged**